src/efi.rs

```diff
@@ -381,7 +402,10 @@ impl Component for Efi {
             .run()?;
         if update_firmware {
             if let Some(vendordir) = self.get_efi_vendor(&Path::new(src_root))? {
-                self.update_firmware(device, destd, &vendordir)?
+                // Only update if EFI firmware variables is readwrite and not empty
+                if self.is_efivars_rw()? {
```

Comment:
I think it'd be cleaner to just do this check in `update_firmware`; then we should be able to assume the efi directory exists and just need to check if it's readonly and no-op then.

Reply:
SGTM, thank you! Also do nothing if the directory is empty (see [bug#2333190](https://bugzilla.redhat.com/show_bug.cgi?id=2333190#c3))